src/app/wallet/shared/signature-address-modal/signature-address-modal.component.ts

```diff
@@ -0,0 +1,155 @@
+
+  openLookup() {
+    const dialogRef = this.dialog.open(AddressLookupComponent);
+    // @TODO confirm lookup type
+    dialogRef.componentInstance.type = 'receive';
+    dialogRef.componentInstance.selectAddressCallback.subscribe((response: AddressLookUpCopy) => {
+      this.selectAddress(response);
+      dialogRef.close();
+    });
+  }
+
+  selectAddress(copyObject: AddressLookUpCopy) {
+    this.formData.address = copyObject.address;
+    this.verifyAddress();
+  }
+
+  // copy code start
+
+  /** verifyAddress: calls RPC to validate it. */
+  verifyAddress() {
+    if (!this.formData.address) {
+      this.validAddress = undefined;
+      return;
+    }
+
+    const validateAddressCB = (response) => {
+       this.validAddress = response.isvalid;
+    };
+
+    this._rpc.call('validateaddress', [this.formData.address])
+      .subscribe(
+        response => validateAddressCB(response),
+        error => this.log.er('verifyAddress: validateAddressCB failed'));
+  }
+  // copy code end
+
+  onFormSubmit(): void {
+    if (this._rpc.state.get('locked')) {
+      // unlock wallet
+      this._modals.open('unlock', {forceOpen: true, timeout: 3, callback: this.signVerifyMessage.bind(this)});
+    } else {
+      // wallet already unlocked
+      this.signVerifyMessage();
+    }
+  }
+
+  signVerifyMessage(): void {
+    const address: string = this.formData.address;
+    const message: string = (this.formData.message !== undefined) ? this.formData.message : '';
+    const signature: string = this.formData.signature;
+
+    if (this.type === 'sign') {
+      this._rpc.call('signmessage', [address, message])
+        .subscribe(response => {
+            this.formData.signature = response;
+            this.flashNotification.open('Message Sign Successfully');
+          },
+          error => {
+          // @TODO add generic message
+            this.flashNotification.open(error.message);
+            this.log.er(`signVerifyMessage, RPC returned an error when signing message. ${error.message}`);
+          });
+    } else {
+      this._rpc.call('verifymessage', [address, signature, message])
+        .subscribe(response => {
+            if (response) {
+              this.flashNotification.open('Message verified.');
+            } else {
+              this.flashNotification.open(
+                `Message verification failed:\n
+                the supplied signature and/or message were invalid for this address!`);
+            }
+          },
+          error => {
+            // @TODO add generic message
+            this.flashNotification.open(error.message);
+            this.log.er(`signVerifyMessage, RPC returned an error when verifying message. ${error.message}`);
+          });
+    }
+  }
+
+  resetForm(): void {
+    this.addressForm.reset();
+  }
+
+  onCopyAddress(): void {
+    this.flashNotification.open('Copied address to clipboard!');
+  }
+
+  pasteAddress(): void {
+    document.getElementById('signVerifyAdd').focus();
```

Comment:
@anandsinghparihar there is a better way to get the element, ViewChild

Reply:
Not sure if it will work with execCommand tho.

Reply:
@kewde execCommand work fine with all cases !!!